### PR TITLE
Fix build issue on Windows

### PR DESCRIPTION
**Pull request by johnsonshih**
conditioned compile to use Windows APIs for generating UUID and Sleep

### CODE REVIEW

File: nodelet/src/nodelet.cpp [hunk 1]
```diff
@@ -341,7 +356,11 @@ int
         ROS_INFO("Bond broken, exiting");
         goto shutdown;
       }
+#ifndef _WIN32
       usleep(100000);
+#else
+      Sleep(100);
```
Comment (mikaelarguedas) [inline nodelet/src/nodelet.cpp:362]: Is there a specific header to include to get this function ?

File: nodelet/src/nodelet.cpp [hunk 2]
```diff
@@ -53,13 +52,29 @@
 #include "nodelet/NodeletLoad.h"
 #include "nodelet/NodeletUnload.h"
 
+#ifndef _WIN32
+#include <uuid/uuid.h>
+#else
+#include <rpc.h>
+#endif
```
Comment (mikaelarguedas) [inline nodelet/src/nodelet.cpp]: please indent the includes for readability (similar to https://github.com/ros/bond_core/pull/38/commits/5ebdaf7b79f060fa546d6f55c69a0e124cff92d5)

**Review by mikaelarguedas — COMMENTED** [review]
changes lgtm, a couple minor comments/questions below

File: nodelet/src/nodelet.cpp (hunk 2 above)
Comment (johnsonshih) [reply nodelet/src/nodelet.cpp]: proper idented

File: nodelet/src/nodelet.cpp (hunk 1 above)
Comment (johnsonshih) [reply nodelet/src/nodelet.cpp:362]: ros/ros.h include ros/platform.h and platform.h include windows.h that has the declare Sleep()

Comment (mikaelarguedas) [reply nodelet/src/nodelet.cpp:362]: great, thanks for confirming

**Review by mikaelarguedas — APPROVED** [review]
lgtm

**Comment by kejxu** [comment]
hi @mikaelarguedas and maintainers, just a friendly ping =) could this change be merged given the approval?

**Comment by mikaelarguedas** [comment]
This still looks good to me.
The melodic test failure seems unrelated to this change and seems to have appeared after lower packages were released in melodic.

Friendly ping @mjcarroll that is the current maintainer

**Review by mjcarroll — APPROVED** [review]
LGTM, thanks for the ping.

**Pull request merged**